adserver/models.py

```diff
@@ -632,14 +632,27 @@ def weighted_clicks_needed_today(self):
         Calculates clicks needed taking into account a flight's priority.
 
         For the purpose of clicks needed, 1000 impressions = 1 click (for CPM ads)
+        Takes into account value of the flight,
+        which causes higher paid and better CTR ads to be prioritized.
         """
         impressions_needed = 0
 
         # This is naive but we are counting a click as being worth 1,000 views
         impressions_needed += math.ceil(self.views_needed_today() / 1000.0)
         impressions_needed += self.clicks_needed_today()
 
-        return impressions_needed * self.priority_multiplier
+        # Cost should be a float from around .5-5, averaging 2
+        cost = float(self.cpm or self.cpc)
+
+        # CTR should be a float around .1
+        # Leading to a base value of $2 * 5 * .1 == 1.0
+        ctr_value = cost * 5.0 * float(self.ctr())
+
+        # Keep values above 1, so we don't penalize the value for lower performance
+        # but add value for higher performance
+        value = max(int(ctr_value), 1)
```

Comment:
I wouldn't mind having a maximum that to this multiplier. Say 10x or something like that. There are weird extremes that can happen like say the first time an ad is shown it's clicked resulting in a CTR of 100%. It's likely to even out pretty quickly but might be worth just capping so we don't run into that.

Also, we should probably keep the value priority multiplier as a float and only cast the final return value to an int.

```suggestion
        value = max(float(ctr_value), 1.0)
        value = min(value, 10.0)
```